app/models/host.rb

```diff
@@ -705,185 +705,6 @@ def verify_credentials_with_ipmi(auth_type = nil)
-    raise _("Starting address is malformed") if (starting =~ pattern).nil?
-    raise _("Ending address is malformed") if (ending =~ pattern).nil?
-
-    starting.split(".").each_index do |i|
-      if starting.split(".")[i].to_i > 255 || ending.split(".")[i].to_i > 255
-        raise _("IP address octets must be 0 to 255")
-      end
-      if starting.split(".")[i].to_i > ending.split(".")[i].to_i
-        raise _("Ending address must be greater than starting address")
-      end
-    end
-
-    network_id = starting.split(".")[0..2].join(".")
-    host_start = starting.split(".").last.to_i
-    host_end = ending.split(".").last.to_i
-
-    host_start.upto(host_end) do |h|
-      ipaddr = network_id + "." + h.to_s
-
-      unless Host.find_by(:ipaddress => ipaddr).nil? # skip discover for existing hosts
-        _log.info("ipaddress '#{ipaddr}' exists, skipping discovery")
-        next
-      end
-
-      discover_options = {:ipaddr         => ipaddr,
-                          :ping           => options[:ping],
-                          :timeout        => options[:timeout],
-                          :discover_types => options[:discover_types],
-                          :credentials    => options[:credentials]
-      }
-
-      # Add Windows domain credentials for HyperV WMI checks
-      default_zone = Zone.find_by(:name => 'default')
-      if !default_zone.nil? && default_zone.has_authentication_type?(:windows_domain)
-        discover_options[:windows_domain] = [default_zone.authentication_userid(:windows_domain), default_zone.authentication_password_encrypted(:windows_domain)]
-      end
-
-      MiqQueue.put(:class_name => "Host", :method_name => "discoverHost", :data => Marshal.dump(discover_options), :server_guid => MiqServer.my_guid)
-    end
-  end
-
-  def reset_discoverable_fields
-    raise _("Host Not Resettable - No IPMI Address") if ipmi_address.blank?
-    cred = authentication_type(:ipmi)
-    raise _("Host Not Resettable - No IPMI Credentials") if cred.nil?
-
-    run_callbacks(:destroy) { false } # Run only the before_destroy callbacks to destroy all associations
-    reload
-
-    attributes.each do |key, _value|
-      next if %w(id guid ipmi_address mac_address name created_on updated_on vmm_vendor).include?(key)
-      send("#{key}=", nil)
-    end
-
-    make_smart # before_create callback
-    self.settings   = nil
-    self.name       = "IPMI <#{ipmi_address}>"
-    self.vmm_vendor = 'unknown'
-    save!
-
-    authentications.create(cred.attributes) unless cred.nil?
-    self
-  end
-
-  def detect_discovered_os(ost)
-    # Determine os
-    os_type = nil
-    if is_vmware?
-      os_name = "VMware ESX Server"
-    elsif ost.os.include?(:linux)
-      os_name = "linux"
-    elsif ost.os.include?(:mswin)
-      os_name = "windows"
-      os_type = os_name
-    else
-      os_name = nil
-    end
-
-    return os_name, os_type
-  end
-
-  def detect_discovered_hypervisor(ost, ipaddr)
```

Comment:
Note there's a subclass override of this method in the vmware repo: https://github.com/ManageIQ/manageiq-providers-vmware/blob/5e3dac8bd0b0ba09a5938ce8b6c5a2d3ace7e860/app/models/manageiq/providers/vmware/infra_manager/host.rb